lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -496,10 +499,29 @@ PathTracker::processPathTrackers(const AnnoTarget &target) {
       } else {
         moduleName = cast<hw::InnerRefAttr>(oldPath.front()).getModule();
       }
+
+      // Copy the leading part of the hierarchical path from the owning module
+      // to the start of the annotation's NLA.
+      InstanceGraphNode *node = instanceGraph.lookup(moduleName);
+      while (true) {
```

Comment:
In this patch, I have just added a fresh instance graph walk up for each hierpath op. We could coordinate this with the walk in getOrComputeAltBasePath, but then we'd need to update the state to cache the array of paths as well. In large designs, I didn't observe any noticeable performance regression from this.